Get all the tests to pass

Why?
--------

Before we start modifying the dtool code to be able to use postgresql, we want to know that we didn't break the system while changing it.  What's an easy way of seeing something is working?  All the tests passing.  The tests need to pass before we make more modifications to the code.

Make this more visible
------------------------------

Run the tests using github workflow.  We'll use this as the standard to avoid differences due to our own development set up.

Current Status
--------------------

I ran the tests on a [wildme/wbia-provision docker container](https://github.com/WildbookOrg/wildbook-ia/blob/develop/devops/provision/Dockerfile):

First to run a container:

```
docker run --rm -ti wildme/wbia-provision
```

In the container:

```
. /virtualenv/env3/bin/activate
cd /wbia/wildbook-ia
mkdir -p data/work
python -m wbia --set-workdir data/work --preload-exit
python _dev/reset_dbs.py
pytest
```

(The tests took about 40 minutes on my machine.)

I got 

```
87 failed, 559 passed, 628 skipped, 41 warnings, 3 errors
```

Then I tried running the tests again without reseting the database:

```
90 failed, 553 passed, 612 skipped, 41 warnings, 3 errors
```

These tests failed with a database reset:

- `wbia/control/manual_annot_funcs.py::get_annot_hashid_semantic_uuid:0`
- `wbia/web/job_engine.py::get_job_metadata:0`

These tests failed without a database reset:

- `wbia/algo/preproc/preproc_occurrence.py::prepare_X_data:0`
- `wbia/control/manual_annot_funcs.py::get_valid_aids:0`
- `wbia/control/manual_gsgrelate_funcs.py::unrelate_images_and_imagesets:0`
- `wbia/control/manual_image_funcs.py::_get_all_image_rowids:0`
- `wbia/control/manual_image_funcs.py::get_valid_gids:0`


<summary>Click here to see the other tests that failed:</summary>
<details>

1. FAILED `wbia/core_images.py::compute_web_src:0`
2. FAILED `wbia/algo/graph/core.py::MiscHelpers.add_aids:0`
3. FAILED `wbia/algo/graph/mixin_dynamic.py::NonDynamicUpdate.apply_nondynamic_update:0`
4. FAILED `wbia/algo/graph/mixin_helpers.py::Convenience.pair_connection_info:0`
5. FAILED `wbia/algo/graph/mixin_helpers.py::DummyEdges.ensure_mst:1`
6. FAILED `wbia/algo/graph/mixin_loops.py::InfrLoops.pos_redun_gen:0`
7. FAILED `wbia/algo/graph/mixin_matching.py::InfrLearning.learn_evaluation_verifiers:0`
8. FAILED `wbia/algo/graph/mixin_wbia.py::IBEISIO.get_wbia_name_delta:1`
9. FAILED `wbia/algo/graph/mixin_wbia.py::IBEISIO.get_wbia_name_delta:2`
10. FAILED `wbia/algo/graph/mixin_wbia.py::IBEISIO.get_wbia_name_delta:3`
11. FAILED `wbia/algo/graph/mixin_wbia.py::IBEISIO.match_state_delta:0`
12. FAILED `wbia/algo/graph/mixin_wbia.py::IBEISIO._make_state_delta:0`
13. FAILED `wbia/algo/graph/mixin_wbia.py::IBEISIO._make_state_delta:1`
14. FAILED `wbia/algo/graph/refresh.py::RefreshCriteria.pred_num_positives:0`
15. FAILED `wbia/algo/graph/refresh.py::demo_refresh:0`
16. FAILED `wbia/algo/graph/tests/test_graph_iden.py::test_incomp_inference - ValueError: cannot apply union_all to an empty list`
17. FAILED `wbia/algo/graph/tests/test_graph_iden.py::test_unrev_inference - ValueError: cannot apply union_all to an empty list`
18. FAILED `wbia/algo/graph/tests/test_graph_iden.py::test_pos_neg - ValueError: cannot apply union_all to an empty list`
19. FAILED `wbia/algo/graph/tests/test_neg_metagraph.py::test_neg_metagraph_simple_add_remove - ValueError: not enough values to unpack (expected 4, ...`                              
20. FAILED `wbia/algo/graph/tests/test_neg_metagraph.py::test_neg_metagraph_merge - AttributeError: 'NiceGraph' object has no attribute 'selfloop_edges'`                              
21. FAILED `wbia/algo/graph/tests/test_neg_metagraph.py::test_neg_metagraph_split_neg - AttributeError: 'NiceGraph' object has no attribute 'selfloop...`                              
22. FAILED `wbia/algo/graph/tests/test_neg_metagraph.py::test_neg_metagraph_split_incomp - AttributeError: 'NiceGraph' object has no attribute 'selfl...`                              
23. FAILED `wbia/algo/graph/tests/test_neg_metagraph.py::test_neg_metagraph_split_and_merge - AttributeError: 'NiceGraph' object has no attribute 'se...`                              
24. FAILED `wbia/algo/hots/chip_match.py::ChipMatch.combine_cms:0`
25. FAILED `wbia/algo/hots/neighbor_index.py::NeighborIndex.knn:0`
26. FAILED `wbia/algo/hots/neighbor_index_cache.py::group_daids_by_cached_nnindexer:0`
27. FAILED `wbia/algo/hots/nn_weights.py::testdata_vn_dists:0`
28. FAILED `wbia/algo/hots/pipeline.py::build_impossible_daids_list:0`
29. FAILED `wbia/algo/smk/inverted_index.py::InvertedAnnots.compute_word_weights:0`
30. FAILED `wbia/algo/smk/inverted_index.py::InvertedAnnots.compute_gammas:0`
31. FAILED `wbia/algo/smk/match_chips5.py::EstimatorRequest.shallowcopy:0`
32. FAILED `wbia/algo/smk/smk_pipeline.py::SMK.match_single:0`
33. FAILED `wbia/algo/verif/clf_helpers.py::ClfProblem.learn_evaluation_classifiers:1`
34. FAILED `wbia/control/manual_annot_funcs.py::get_annot_hashid_semantic_uuid:0`
35. FAILED `wbia/control/manual_annot_funcs.py::get_annot_semantic_uuids:0`
36. FAILED `wbia/control/manual_annot_funcs.py::get_annot_visual_uuids:0`
37. FAILED `wbia/control/manual_chip_funcs.py::get_annot_chips:0`
38. FAILED `wbia/control/manual_chip_funcs.py::get_part_chips:0`
39. FAILED `wbia/control/manual_feat_funcs.py::get_annot_num_feats:0`
40. FAILED `wbia/control/manual_image_funcs.py::add_images:1`
41. FAILED `wbia/control/manual_image_funcs.py::localize_images:0`
42. FAILED `wbia/control/manual_image_funcs.py::get_image_uuids:0`
43. FAILED `wbia/control/manual_image_funcs.py::get_image_hash:0`
44. FAILED `wbia/control/manual_image_funcs.py::get_image_gnames:0`
45. FAILED `wbia/control/manual_imageset_funcs.py::get_imageset_num_imgs_reviewed:0`
46. FAILED `wbia/control/manual_imageset_funcs.py::get_imageset_num_annots_reviewed:0`
47. FAILED `wbia/control/manual_imageset_funcs.py::get_imageset_nids:0`
48. FAILED `wbia/control/manual_imageset_funcs.py::get_imageset_name_uuids:0`
49. FAILED `wbia/control/manual_imageset_funcs.py::update_imageset_info:0`
50. FAILED `wbia/control/wildbook_manager.py::update_wildbook_install_config:0`
51. FAILED `wbia/dbio/export_hsdb.py::get_hots_table_strings:0`
52. FAILED `wbia/dtool/_grave_depcache.py::get_all_ancestor_rowids:0`
53. FAILED `wbia/gui/id_review_api.py::ensure_match_img:0`
54. FAILED `wbia/gui/id_review_api.py::make_ensure_match_img_nosql_func:0`
55. FAILED `wbia/guitool/api_item_widget.py::simple_api_tree_widget:0`
56. FAILED `wbia/guitool/api_table_view.py::APITableView._init_header_behavior:0`
57. FAILED `wbia/guitool/api_table_view.py::APITableView.keyPressEvent:0`
58. FAILED `wbia/guitool/api_thumb_delegate.py::read_thumb_as_qimg:0`
59. FAILED `wbia/guitool/guitool_components.py::ProgHook:0`
60. FAILED `wbia/guitool/guitool_components.py::newLabel:0`
61. FAILED `wbia/guitool/guitool_components.py::Spoiler:0`
62. FAILED `wbia/guitool/guitool_dialogs.py::msgbox:0`
63. FAILED `wbia/other/dbinfo.py::get_short_infostr:0`
64. FAILED `wbia/other/ibsfuncs.py::get_consecutive_newname_list_via_species:1`
65. FAILED `wbia/plottool/nx_helpers.py::nx_agraph_layout:0`
66. FAILED `wbia/tests/reset_testdbs.py::reset_mtest:0`
67. FAILED `wbia/unstable/bayes.py::reduce_marginalize:0 - RuntimeError: ERROR: Failed to import modname=bayes with modpath=/wbia/wildbook-ia/wbia/un...`                              
68. FAILED `wbia/unstable/demobayes.py::get_toy_data_1v1:0 - RuntimeError: ERROR: Failed to import modname=demobayes with modpath=/wbia/wildbook-ia/w...`                              
69. FAILED `wbia/unstable/demobayes.py::learn_prob_score:0 - RuntimeError: ERROR: Failed to import modname=demobayes with modpath=/wbia/wildbook-ia/w...`                              
70. FAILED `wbia/unstable/old_dbinfo_dan.py::tst_name_consistency:0`
71. FAILED `wbia/unstable/old_scorers.py::cos_match_weighter:0 - RuntimeError: ERROR: Failed to import modname=old_scorers with modpath=/wbia/wildboo...`                              
72. FAILED `wbia/unstable/orig_graph_iden.py::OrigAnnotInference:0`
73. FAILED `wbia/unstable/orig_graph_iden.py::OrigAnnotInference:1`
74. FAILED `wbia/unstable/orig_graph_iden.py::OrigAnnotInference:2`
75. FAILED `wbia/unstable/scorenorm.py::get_global_species_scorenorm_cachedir:0`
76. FAILED `wbia/unstable/scorenorm.py::learn_annotscore_normalizer:0`
77. FAILED `wbia/unstable/scorenorm.py::learn_featscore_normalizer:0`
78. FAILED `wbia/unstable/scorenorm.py::get_training_featscores:0`
79. FAILED `wbia/unstable/scorenorm.py::get_topannot_training_idxs:0`
80. FAILED `wbia/unstable/scorenorm.py::get_topname_training_idxs:0`
81. FAILED `wbia/unstable/scorenorm.py::get_training_fsv:0`
82. FAILED `wbia/unstable/scorenorm.py::get_training_desc_dist:0`
83. FAILED `wbia/viz/viz_qres.py::show_qres_analysis:0`
84. FAILED `wbia/viz/viz_qres.py::show_qres_analysis:1`
85. FAILED `wbia/web/apis.py::annot_src_api:0`
86. FAILED `wbia/web/apis.py::background_src_api:0`
87. ERROR `wbia/plottool/tests/test_interact_multi_image.py::test_interact_multimage`
88. ERROR `wbia/plottool/tests/test_viz_image2.py::test_viz_image`
89. ERROR `wbia/plottool/tests/test_viz_images.py::test_viz_image`



</details>

I'm going to look into

```
FAILED wbia/unstable/bayes.py::reduce_marginalize:0 - RuntimeError: ERROR: Failed to import modname=bayes with modpath=/wbia/wildbook-ia/wbia/un...
FAILED wbia/unstable/demobayes.py::get_toy_data_1v1:0 - RuntimeError: ERROR: Failed to import modname=demobayes with modpath=/wbia/wildbook-ia/w...
FAILED wbia/unstable/demobayes.py::learn_prob_score:0 - RuntimeError: ERROR: Failed to import modname=demobayes with modpath=/wbia/wildbook-ia/w...
```

<summary>This is the current list of failed tests: (142 failed, 504 passed, 628 skipped, 17 warnings) </summary>

<details>

```
============================================================= short test summary info =============================================================
FAILED wbia/core_images.py::compute_web_src:0
FAILED wbia/algo/graph/core.py::MiscHelpers.add_aids:0
FAILED wbia/algo/graph/mixin_dynamic.py::NonDynamicUpdate.apply_nondynamic_update:0
FAILED wbia/algo/graph/mixin_helpers.py::Convenience.pair_connection_info:0
FAILED wbia/algo/graph/mixin_loops.py::InfrLoops.main_gen:0
FAILED wbia/algo/graph/mixin_matching.py::AnnotInfrMatching._cm_training_pairs:0
FAILED wbia/algo/graph/mixin_matching.py::AnnotInfrMatching.apply_match_scores:0
FAILED wbia/algo/graph/mixin_matching.py::CandidateSearch.find_lnbnn_candidate_edges:0
FAILED wbia/algo/graph/mixin_wbia.py::IBEISIO.match_state_delta:0
FAILED wbia/algo/graph/mixin_wbia.py::IBEISIO._make_state_delta:0
FAILED wbia/algo/graph/mixin_wbia.py::IBEISIO._make_state_delta:1
FAILED wbia/algo/graph/refresh.py::RefreshCriteria.pred_num_positives:0
FAILED wbia/algo/graph/refresh.py::demo_refresh:0
FAILED wbia/algo/graph/tests/test_graph_iden.py::test_incomp_inference - ValueError: cannot apply union_all to an empty list
FAILED wbia/algo/graph/tests/test_graph_iden.py::test_unrev_inference - ValueError: cannot apply union_all to an empty list
FAILED wbia/algo/graph/tests/test_graph_iden.py::test_pos_neg - ValueError: cannot apply union_all to an empty list
FAILED wbia/algo/graph/tests/test_neg_metagraph.py::test_neg_metagraph_simple_add_remove - ValueError: not enough values to unpack (expected 4, ...
FAILED wbia/algo/graph/tests/test_neg_metagraph.py::test_neg_metagraph_merge - AttributeError: 'NiceGraph' object has no attribute 'selfloop_edges'
FAILED wbia/algo/graph/tests/test_neg_metagraph.py::test_neg_metagraph_split_neg - AttributeError: 'NiceGraph' object has no attribute 'selfloop...
FAILED wbia/algo/graph/tests/test_neg_metagraph.py::test_neg_metagraph_split_incomp - AttributeError: 'NiceGraph' object has no attribute 'selfl...
FAILED wbia/algo/graph/tests/test_neg_metagraph.py::test_neg_metagraph_split_and_merge - AttributeError: 'NiceGraph' object has no attribute 'se...
FAILED wbia/algo/hots/chip_match.py::_ChipMatchVisualization.show_single_namematch:0
FAILED wbia/algo/hots/chip_match.py::_ChipMatchVisualization.show_single_annotmatch:0
FAILED wbia/algo/hots/chip_match.py::_ChipMatchVisualization.show_single_annotmatch:1
FAILED wbia/algo/hots/chip_match.py::_ChipMatchVisualization.ishow_analysis:0
FAILED wbia/algo/hots/chip_match.py::_ChipMatchScorers.evaluate_csum_annot_score:0
FAILED wbia/algo/hots/chip_match.py::_ChipMatchScorers.score_annot_csum:0
FAILED wbia/algo/hots/chip_match.py::_ChipMatchScorers.score_name_nsum:0
FAILED wbia/algo/hots/chip_match.py::MatchBaseIO.save_to_fpath:0
FAILED wbia/algo/hots/chip_match.py::_AnnotMatchConvenienceGetter.get_name_shortlist_aids:0
FAILED wbia/algo/hots/chip_match.py::_AnnotMatchConvenienceGetter.get_annot_shortlist_aids:0
FAILED wbia/algo/hots/chip_match.py::_ChipMatchDebugger.get_inspect_str:0
FAILED wbia/algo/hots/chip_match.py::_ChipMatchDebugger.get_rawinfostr:0
FAILED wbia/algo/hots/chip_match.py::_ChipMatchDebugger.get_cvs_str:0
FAILED wbia/algo/hots/chip_match.py::ChipMatch.extend_results:0
FAILED wbia/algo/hots/chip_match.py::ChipMatch.combine_cms:0
FAILED wbia/algo/hots/chip_match.py::ChipMatch.take_annots:0
FAILED wbia/algo/hots/chip_match.py::ChipMatch.take_annots:1
FAILED wbia/algo/hots/chip_match.py::ChipMatch.take_feature_matches:0
FAILED wbia/algo/hots/chip_match.py::ChipMatch.from_json:0
FAILED wbia/algo/hots/chip_match.py::ChipMatch.to_json:0
FAILED wbia/algo/hots/chip_match.py::ChipMatch.to_json:1
FAILED wbia/algo/hots/chip_match.py::get_chipmatch_fname:0
FAILED wbia/algo/hots/name_scoring.py::get_chipmatch_namescore_nonvoting_feature_flags:0
FAILED wbia/algo/hots/name_scoring.py::align_name_scores_with_annots:0
FAILED wbia/algo/hots/neighbor_index.py::NeighborIndex:0
FAILED wbia/algo/hots/neighbor_index.py::NeighborIndex.knn:0
FAILED wbia/algo/hots/neighbor_index.py::NeighborIndex.requery_knn:0
FAILED wbia/algo/hots/neighbor_index.py::NeighborIndex.get_nn_aids:0
FAILED wbia/algo/hots/neighbor_index.py::NeighborIndex.get_nn_fgws:0
FAILED wbia/algo/hots/neighbor_index_cache.py::request_wbia_nnindexer:0
FAILED wbia/algo/hots/neighbor_index_cache.py::request_augmented_wbia_nnindexer:0
FAILED wbia/algo/hots/neighbor_index_cache.py::group_daids_by_cached_nnindexer:0
FAILED wbia/algo/hots/neighbor_index_cache.py::testdata_nnindexer:0
FAILED wbia/algo/hots/nn_weights.py::fg_match_weighter:0
FAILED wbia/algo/hots/nn_weights.py::nn_normalized_weight:0
FAILED wbia/algo/hots/nn_weights.py::nn_normalized_weight:1
FAILED wbia/algo/hots/nn_weights.py::get_normk:0
FAILED wbia/algo/hots/nn_weights.py::apply_normweight:0
FAILED wbia/algo/hots/nn_weights.py::get_name_normalizers:0
FAILED wbia/algo/hots/nn_weights.py::testdata_vn_dists:0
FAILED wbia/algo/hots/nn_weights.py::all_normalized_weights_test:0
FAILED wbia/algo/hots/pipeline.py::build_impossible_daids_list:0
FAILED wbia/algo/hots/pipeline.py::nearest_neighbor_cacheid2:0
FAILED wbia/algo/hots/pipeline.py::nearest_neighbors:0
FAILED wbia/algo/hots/pipeline.py::nearest_neighbors:1
FAILED wbia/algo/hots/pipeline.py::nearest_neighbors:2
FAILED wbia/algo/hots/pipeline.py::nearest_neighbors:3
FAILED wbia/algo/hots/pipeline.py::baseline_neighbor_filter:0
FAILED wbia/algo/hots/pipeline.py::weight_neighbors:0
FAILED wbia/algo/hots/pipeline.py::weight_neighbors:1
FAILED wbia/algo/hots/pipeline.py::spatial_verification:0
FAILED wbia/algo/hots/pipeline.py::compute_matching_dlen_extent:0
FAILED wbia/algo/hots/requery_knn.py::requery_knn:0
FAILED wbia/algo/hots/scoring.py::make_chipmatch_shortlists:0
FAILED wbia/algo/hots/tests/test_lnbnn.py::test_lnbnn - TypeError: has_key() missing 1 required positional argument: 'item'
FAILED wbia/algo/hots/tests/test_scoremech.py::test_scoremech - TypeError: has_key() missing 1 required positional argument: 'item'
FAILED wbia/algo/smk/inverted_index.py::InvertedAnnots.compute_word_weights:0
FAILED wbia/algo/smk/inverted_index.py::InvertedAnnots.compute_gammas:0
FAILED wbia/algo/smk/match_chips5.py::EstimatorRequest.shallowcopy:0
FAILED wbia/algo/smk/smk_pipeline.py::SMK.match_single:0
FAILED wbia/algo/verif/clf_helpers.py::ClfProblem.learn_evaluation_classifiers:1
FAILED wbia/control/manual_annot_funcs.py::get_annot_hashid_semantic_uuid:0
FAILED wbia/control/manual_annot_funcs.py::get_annot_semantic_uuids:0
FAILED wbia/control/manual_annot_funcs.py::get_annot_visual_uuids:0
FAILED wbia/control/manual_chip_funcs.py::get_annot_chips:0
FAILED wbia/control/manual_chip_funcs.py::get_part_chips:0
FAILED wbia/control/manual_feat_funcs.py::get_annot_num_feats:0
FAILED wbia/control/manual_image_funcs.py::add_images:1
FAILED wbia/control/manual_image_funcs.py::localize_images:0
FAILED wbia/control/manual_image_funcs.py::get_image_uuids:0
FAILED wbia/control/manual_image_funcs.py::get_image_hash:0
FAILED wbia/control/manual_image_funcs.py::get_image_gnames:0
FAILED wbia/control/manual_imageset_funcs.py::get_imageset_num_imgs_reviewed:0
FAILED wbia/control/manual_imageset_funcs.py::get_imageset_num_annots_reviewed:0
FAILED wbia/control/manual_imageset_funcs.py::get_imageset_nids:0
FAILED wbia/control/manual_imageset_funcs.py::get_imageset_name_uuids:0
FAILED wbia/control/manual_imageset_funcs.py::update_imageset_info:0
FAILED wbia/control/wildbook_manager.py::update_wildbook_install_config:0
FAILED wbia/dbio/export_hsdb.py::get_hots_table_strings:0
FAILED wbia/dtool/_grave_depcache.py::get_all_ancestor_rowids:0
FAILED wbia/expt/test_result.py::build_cmsinfo:0
FAILED wbia/expt/test_result.py::TestResult.get_cfgx_groupxs:0
FAILED wbia/expt/test_result.py::TestResult.get_all_varied_params:0
FAILED wbia/expt/test_result.py::TestResult.make_figtitle:0
FAILED wbia/expt/test_result.py::TestResult.get_truth2_prop:0
FAILED wbia/gui/id_review_api.py::make_review_api:0
FAILED wbia/gui/id_review_api.py::ensure_match_img:0
FAILED wbia/gui/id_review_api.py::make_ensure_match_img_nosql_func:0
FAILED wbia/gui/inspect_gui.py::get_aidpair_context_menu_options:0
FAILED wbia/guitool/api_table_view.py::APITableView._init_header_behavior:0
FAILED wbia/guitool/api_table_view.py::APITableView.keyPressEvent:0
FAILED wbia/guitool/api_thumb_delegate.py::read_thumb_as_qimg:0
FAILED wbia/guitool/guitool_components.py::newLabel:0
FAILED wbia/guitool/guitool_components.py::Spoiler:0
FAILED wbia/guitool/guitool_dialogs.py::msgbox:0
FAILED wbia/init/main_helpers.py::testdata_cm:0
FAILED wbia/other/dbinfo.py::get_short_infostr:0
FAILED wbia/other/ibsfuncs.py::get_consecutive_newname_list_via_species:1
FAILED wbia/plottool/interact_matches.py::MatchInteraction2:0
FAILED wbia/plottool/nx_helpers.py::nx_agraph_layout:0
FAILED wbia/tests/reset_testdbs.py::reset_mtest:0
FAILED wbia/unstable/bayes.py::reduce_marginalize:0
FAILED wbia/unstable/old_dbinfo_dan.py::tst_name_consistency:0
FAILED wbia/unstable/old_scorers.py::cos_match_weighter:0
FAILED wbia/unstable/orig_graph_iden.py::OrigAnnotInference:0
FAILED wbia/unstable/orig_graph_iden.py::OrigAnnotInference:1
FAILED wbia/unstable/orig_graph_iden.py::OrigAnnotInference:2
FAILED wbia/unstable/scorenorm.py::get_global_species_scorenorm_cachedir:0
FAILED wbia/unstable/scorenorm.py::learn_annotscore_normalizer:0
FAILED wbia/unstable/scorenorm.py::learn_featscore_normalizer:0
FAILED wbia/unstable/scorenorm.py::get_training_featscores:0
FAILED wbia/unstable/scorenorm.py::get_topannot_training_idxs:0
FAILED wbia/unstable/scorenorm.py::get_topname_training_idxs:0
FAILED wbia/unstable/scorenorm.py::get_training_fsv:0
FAILED wbia/unstable/scorenorm.py::get_training_desc_dist:0
FAILED wbia/viz/viz_qres.py::show_qres_analysis:0
FAILED wbia/viz/viz_qres.py::show_qres_analysis:1
FAILED wbia/viz/interact/interact_matches.py::testdata_match_interact:0
FAILED wbia/web/apis.py::annot_src_api:0
FAILED wbia/web/apis.py::background_src_api:0
FAILED wbia/web/job_engine.py::get_job_metadata:0
===================================== 142 failed, 504 passed, 628 skipped, 17 warnings in 2594.59s (0:43:14) ======================================
```

</details>

With #59 and a temporary fix for https://github.com/WildbookOrg/wildbook-ia/issues/51#issuecomment-658962680

```diff
diff --git a/wbia/dbio/ingest_hsdb.py b/wbia/dbio/ingest_hsdb.py
index c4196002d..d3a1ae0dc 100644
--- a/wbia/dbio/ingest_hsdb.py
+++ b/wbia/dbio/ingest_hsdb.py
@@ -16,7 +16,7 @@ import csv
 print, rrr, profile = ut.inject2(__name__)
 
 
-SUCCESS_FLAG_FNAME = '_hsdb_to_wbia_convert_success'
+SUCCESS_FLAG_FNAME = '_hsdb_to_ibeis_convert_success'
 
 
 def is_hsdb(dbdir):
```

We are back down to `85 failed, 561 passed, 628 skipped, 17 warnings in 2238.20s (0:37:18)`

```
============================================================= short test summary info =============================================================
FAILED wbia/core_images.py::compute_web_src:0
FAILED wbia/algo/graph/core.py::MiscHelpers.add_aids:0
FAILED wbia/algo/graph/mixin_dynamic.py::NonDynamicUpdate.apply_nondynamic_update:0
FAILED wbia/algo/graph/mixin_helpers.py::Convenience.pair_connection_info:0
FAILED wbia/algo/graph/mixin_helpers.py::DummyEdges.ensure_mst:1
FAILED wbia/algo/graph/mixin_loops.py::InfrLoops.pos_redun_gen:0
FAILED wbia/algo/graph/mixin_matching.py::InfrLearning.learn_evaluation_verifiers:0
FAILED wbia/algo/graph/mixin_wbia.py::IBEISIO.get_wbia_name_delta:1
FAILED wbia/algo/graph/mixin_wbia.py::IBEISIO.get_wbia_name_delta:2
FAILED wbia/algo/graph/mixin_wbia.py::IBEISIO.get_wbia_name_delta:3
FAILED wbia/algo/graph/mixin_wbia.py::IBEISIO.match_state_delta:0
FAILED wbia/algo/graph/mixin_wbia.py::IBEISIO._make_state_delta:0
FAILED wbia/algo/graph/mixin_wbia.py::IBEISIO._make_state_delta:1
FAILED wbia/algo/graph/refresh.py::RefreshCriteria.pred_num_positives:0
FAILED wbia/algo/graph/refresh.py::demo_refresh:0
FAILED wbia/algo/graph/tests/test_graph_iden.py::test_incomp_inference - ValueError: cannot apply union_all to an empty list
FAILED wbia/algo/graph/tests/test_graph_iden.py::test_unrev_inference - ValueError: cannot apply union_all to an empty list
FAILED wbia/algo/graph/tests/test_graph_iden.py::test_pos_neg - ValueError: cannot apply union_all to an empty list
FAILED wbia/algo/graph/tests/test_neg_metagraph.py::test_neg_metagraph_simple_add_remove - ValueError: not enough values to unpack (expected 4, ...
FAILED wbia/algo/graph/tests/test_neg_metagraph.py::test_neg_metagraph_merge - AttributeError: 'NiceGraph' object has no attribute 'selfloop_edges'
FAILED wbia/algo/graph/tests/test_neg_metagraph.py::test_neg_metagraph_split_neg - AttributeError: 'NiceGraph' object has no attribute 'selfloop...
FAILED wbia/algo/graph/tests/test_neg_metagraph.py::test_neg_metagraph_split_incomp - AttributeError: 'NiceGraph' object has no attribute 'selfl...
FAILED wbia/algo/graph/tests/test_neg_metagraph.py::test_neg_metagraph_split_and_merge - AttributeError: 'NiceGraph' object has no attribute 'se...
FAILED wbia/algo/hots/chip_match.py::ChipMatch.combine_cms:0
FAILED wbia/algo/hots/neighbor_index.py::NeighborIndex.knn:0
FAILED wbia/algo/hots/neighbor_index_cache.py::group_daids_by_cached_nnindexer:0
FAILED wbia/algo/hots/nn_weights.py::testdata_vn_dists:0
FAILED wbia/algo/hots/pipeline.py::build_impossible_daids_list:0
FAILED wbia/algo/smk/inverted_index.py::InvertedAnnots.compute_word_weights:0
FAILED wbia/algo/smk/inverted_index.py::InvertedAnnots.compute_gammas:0
FAILED wbia/algo/smk/match_chips5.py::EstimatorRequest.shallowcopy:0
FAILED wbia/algo/smk/smk_pipeline.py::SMK.match_single:0
FAILED wbia/algo/verif/clf_helpers.py::ClfProblem.learn_evaluation_classifiers:1
FAILED wbia/control/manual_annot_funcs.py::get_annot_hashid_semantic_uuid:0
FAILED wbia/control/manual_annot_funcs.py::get_annot_semantic_uuids:0
FAILED wbia/control/manual_annot_funcs.py::get_annot_visual_uuids:0
FAILED wbia/control/manual_chip_funcs.py::get_annot_chips:0
FAILED wbia/control/manual_chip_funcs.py::get_part_chips:0
FAILED wbia/control/manual_feat_funcs.py::get_annot_num_feats:0
FAILED wbia/control/manual_image_funcs.py::add_images:1
FAILED wbia/control/manual_image_funcs.py::localize_images:0
FAILED wbia/control/manual_image_funcs.py::get_image_uuids:0
FAILED wbia/control/manual_image_funcs.py::get_image_hash:0
FAILED wbia/control/manual_image_funcs.py::get_image_gnames:0
FAILED wbia/control/manual_imageset_funcs.py::get_imageset_num_imgs_reviewed:0
FAILED wbia/control/manual_imageset_funcs.py::get_imageset_num_annots_reviewed:0
FAILED wbia/control/manual_imageset_funcs.py::get_imageset_nids:0
FAILED wbia/control/manual_imageset_funcs.py::get_imageset_name_uuids:0
FAILED wbia/control/manual_imageset_funcs.py::update_imageset_info:0
FAILED wbia/control/wildbook_manager.py::update_wildbook_install_config:0
FAILED wbia/dbio/export_hsdb.py::get_hots_table_strings:0
FAILED wbia/dtool/_grave_depcache.py::get_all_ancestor_rowids:0
FAILED wbia/gui/id_review_api.py::ensure_match_img:0
FAILED wbia/gui/id_review_api.py::make_ensure_match_img_nosql_func:0
FAILED wbia/guitool/api_table_view.py::APITableView._init_header_behavior:0
FAILED wbia/guitool/api_table_view.py::APITableView.keyPressEvent:0
FAILED wbia/guitool/api_thumb_delegate.py::read_thumb_as_qimg:0
FAILED wbia/guitool/guitool_components.py::newLabel:0
FAILED wbia/guitool/guitool_components.py::Spoiler:0
FAILED wbia/guitool/guitool_dialogs.py::msgbox:0
FAILED wbia/other/dbinfo.py::get_short_infostr:0
FAILED wbia/other/ibsfuncs.py::get_consecutive_newname_list_via_species:0
FAILED wbia/other/ibsfuncs.py::get_consecutive_newname_list_via_species:1
FAILED wbia/other/ibsfuncs.py::make_next_name:0
FAILED wbia/plottool/nx_helpers.py::nx_agraph_layout:0
FAILED wbia/tests/reset_testdbs.py::reset_mtest:0
FAILED wbia/unstable/bayes.py::reduce_marginalize:0
FAILED wbia/unstable/old_dbinfo_dan.py::tst_name_consistency:0
FAILED wbia/unstable/old_scorers.py::cos_match_weighter:0
FAILED wbia/unstable/orig_graph_iden.py::OrigAnnotInference:0
FAILED wbia/unstable/orig_graph_iden.py::OrigAnnotInference:1
FAILED wbia/unstable/orig_graph_iden.py::OrigAnnotInference:2
FAILED wbia/unstable/scorenorm.py::get_global_species_scorenorm_cachedir:0
FAILED wbia/unstable/scorenorm.py::learn_annotscore_normalizer:0
FAILED wbia/unstable/scorenorm.py::learn_featscore_normalizer:0
FAILED wbia/unstable/scorenorm.py::get_training_featscores:0
FAILED wbia/unstable/scorenorm.py::get_topannot_training_idxs:0
FAILED wbia/unstable/scorenorm.py::get_topname_training_idxs:0
FAILED wbia/unstable/scorenorm.py::get_training_fsv:0
FAILED wbia/unstable/scorenorm.py::get_training_desc_dist:0
FAILED wbia/viz/viz_qres.py::show_qres_analysis:0
FAILED wbia/viz/viz_qres.py::show_qres_analysis:1
FAILED wbia/web/apis.py::annot_src_api:0
FAILED wbia/web/apis.py::background_src_api:0
FAILED wbia/web/job_engine.py::get_job_metadata:0
====================================== 85 failed, 561 passed, 628 skipped, 17 warnings in 2238.20s (0:37:18) ======================================
```

--- a/wbia/algo/graph/mixin_matching.py
+++ b/wbia/algo/graph/mixin_matching.py
@@ -447,6 +447,7 @@ class InfrLearning(object):
             python -m wbia.algo.graph.mixin_matching learn_evaluation_verifiers
 
         Doctest:
+            >>> # xdoctest: +REQUIRES(module:wbia_cnn)
             >>> import wbia
             >>> infr = wbia.AnnotInference(
             >>>     'PZ_MTEST', aids='all', autoinit='annotmatch',

--- a/wbia/algo/graph/mixin_viz.py
+++ b/wbia/algo/graph/mixin_viz.py
@@ -695,6 +695,7 @@ class GraphVisualization(object):
             python -m wbia.algo.graph.mixin_viz GraphVisualization.show_graph --show
 
         Example:
+            >>> # xdoctest: +REQUIRES(module:pygraphviz)
             >>> # ENABLE_DOCTEST
             >>> from wbia.algo.graph.mixin_viz import *  # NOQA
             >>> from wbia.algo.graph import demo

--- a/wbia/algo/graph/mixin_wbia.py
+++ b/wbia/algo/graph/mixin_wbia.py
@@ -798,15 +798,16 @@ class IBEISIO(object):
             >>> edge_delta_df = infr.match_state_delta()
             >>> subset = edge_delta_df[['old_evidence_decision', 'new_evidence_decision']]
             >>> result = str(subset)
-            >>> # if this doctest fails maybe PZ_MTEST has a non-determenistic reset?
+            >>> # sort result by aid1
+            >>> result = '\n'.join(result.splitlines()[:2] + sorted(result.splitlines()[2:]))
             >>> print(result)
                       old_evidence_decision new_evidence_decision
             aid1 aid2
             1    2                    match               nomatch
+            4    5                      NaN                 match
+            5    8               unreviewed               nomatch
             6    7               unreviewed               nomatch
             7    8                    match                 match
-            5    8               unreviewed               nomatch
-            4    5                      NaN                 match
         """
         old_feedback = infr._feedback_df(old)
         new_feedback = infr._feedback_df(new)
@@ -858,16 +859,17 @@ class IBEISIO(object):
             >>> edge_delta_df = AnnotInference._make_state_delta(old_feedback,
             >>>                                                  new_feedback)
             >>> result = ('edge_delta_df =\n%s' % (edge_delta_df,))
-            >>> print(result)
+            >>> # Change all the "nan" to "NaN"
+            >>> print(result.replace('nan', 'NaN'))
             edge_delta_df =
                        am_rowid old_evidence_decision  ... new_meta_decision  is_new
             aid1 aid2                                  ...
             101  102  1001.0...                 match  ...              null   False
             103  104  1002.0...                 match  ...              null   False
             101  104  1004.0...               nomatch  ...              null   False
-            100  103        nan                   NaN  ...              null    True
-            102  103        nan                   NaN  ...              null    True
-            107  109        nan                   NaN  ...              same    True
+            100  103        NaN                   NaN  ...              null    True
+            102  103        NaN                   NaN  ...              null    True
+            107  109        NaN                   NaN  ...              same    True
             ...
         """
         import wbia

--- a/wbia/algo/hots/neighbor_index.py
+++ b/wbia/algo/hots/neighbor_index.py
@@ -34,7 +34,7 @@ def get_support_data(qreq_, daid_list):
         python -m wbia.algo.hots.neighbor_index get_support_data --show
 
     Example:
-        >>> # ENABLE_DOCTEST
+        >>> # xdoctest: +REQUIRES(module:wbia_cnn)
         >>> from wbia.algo.hots.neighbor_index import *  # NOQA
         >>> import wbia
         >>> qreq_ = wbia.testdata_qreq_(defaultdb='PZ_MTEST', p=':fgw_thresh=.9,maxscale_thresh=10', a=':size=2')

--- a/wbia/algo/hots/nn_weights.py
+++ b/wbia/algo/hots/nn_weights.py
@@ -99,7 +99,7 @@ def fg_match_weighter(nns_list, nnvalid0_list, qreq_):
         python -m wbia.algo.hots.nn_weights --exec-fg_match_weighter
 
     Example:
-        >>> # ENABLE_DOCTEST
+        >>> # xdoctest: +REQUIRES(module:wbia_cnn)
         >>> from wbia.algo.hots.nn_weights import *  # NOQA
         >>> #tup = plh.testdata_pre_weight_neighbors('PZ_MTEST')
         >>> #ibs, qreq_, nns_list, nnvalid0_list = tup

--- a/wbia/algo/verif/deploy.py
+++ b/wbia/algo/verif/deploy.py
@@ -451,6 +451,7 @@ class Deployer(object):
                 * PUBLISH TO /media/hdd/PUBLIC/models/pairclf
 
         Example:
+            >>> # xdoctest: +REQUIRES(module:wbia_cnn)
             >>> from wbia.algo.verif.vsone import *  # NOQA
             >>> params = dict(sample_method='random')
             >>> pblm = OneVsOneProblem.from_empty('PZ_MTEST', **params)

--- a/wbia/algo/verif/pairfeat.py
+++ b/wbia/algo/verif/pairfeat.py
@@ -295,6 +295,7 @@ class PairwiseFeatureExtractor(object):
         object (feature corresopndences) doesnt directly provide.
 
         Example:
+            >>> # xdoctest: +REQUIRES(module:wbia_cnn)
             >>> # ENABLE_DOCTEST
             >>> from wbia.algo.verif.pairfeat import *  # NOQA
             >>> import wbia
@@ -346,6 +347,7 @@ class PairwiseFeatureExtractor(object):
             python -m wbia.algo.verif.pairfeat _make_pairwise_features
 
         Doctest:
+            >>> # xdoctest: +REQUIRES(module:wbia_cnn)
             >>> from wbia.algo.verif.pairfeat import *
             >>> from wbia.algo.graph import demo
             >>> infr = demo.demodata_mtest_infr()

--- a/wbia/algo/verif/vsone.py
+++ b/wbia/algo/verif/vsone.py
@@ -71,7 +71,7 @@ class OneVsOneProblem(clf_helpers.ClfProblem):
         python -m wbia.algo.verif.vsone evaluate_classifiers --db testdb1 --show -a default
 
     Example:
-            >>> # DISABLE_DOCTEST
+        >>> # xdoctest: +REQUIRES(module:wbia_cnn)
         >>> from wbia.algo.verif.vsone import *  # NOQA
         >>> pblm = OneVsOneProblem.from_empty('PZ_MTEST')
         >>> pblm.hyper_params['xval_kw']['n_splits'] = 10

--- a/wbia/control/manual_featweight_funcs.py
+++ b/wbia/control/manual_featweight_funcs.py
@@ -24,6 +24,7 @@ def get_annot_fgweights(ibs, aid_list, config2_=None, ensure=True):
         python -m wbia.control.manual_featweight_funcs get_annot_fgweights
 
     Example:
+        >>> # xdoctest: +REQUIRES(module:wbia_cnn)
         >>> # ENABLE_DOCTEST
         >>> from wbia.control.manual_featweight_funcs import *  # NOQA
         >>> import wbia

--- a/wbia/viz/viz_chip.py
+++ b/wbia/viz/viz_chip.py
@@ -153,6 +153,7 @@ def show_chip(
         python -m wbia.viz.viz_chip show_chip --show --db PZ_MTEST --aid 1 --ecc --draw_lbls=False --notitle --save=~/slides/lnbnn_query.jpg --dpi=300
 
     Example:
+        >>> # xdoctest: +REQUIRES(module:wbia_cnn)
         >>> # VIZ_TEST
         >>> from wbia.viz.viz_chip import *  # NOQA
         >>> import numpy as np

--- a/wbia/web/graph_server.py
+++ b/wbia/web/graph_server.py
@@ -124,6 +124,7 @@ class GraphActor(GRAPH_ACTOR_CLASS):
 
 
     Doctest:
+        >>> # xdoctest: +REQUIRES(module:wbia_cnn)
         >>> from wbia.web.graph_server import *
         >>> import wbia
         >>> actor = GraphActor()

